### DISCUSSION TITLE
Custom tab styles?

### DISCUSSION THREAD

**Discussion by lokesh-krishna** [kovidgoyal/kitty#3984] [Ideas]
is it possible to have custom tab styles or modify the existing ones? I'm presently using the fade option but without any actual face, instead using tab_fade to create margins between the tabs:

![image](https://user-images.githubusercontent.com/20235646/131288856-dd755d37-c0af-492a-a327-4d5972162f50.png)

I was wondering if I could set the characters for the margins of these boxes so that I could make them rounded, something like what I'm doing with my vim status line:

![image](https://user-images.githubusercontent.com/20235646/131288919-917f23f8-063a-4a05-96e3-525e6e445cdb.png)

**Comment by kovidgoyal** [comment]
Use the powerline style for the tab separator characters. There are
extensive customization possibilities, see
https://sw.kovidgoyal.net/kitty/conf/ 

And in the next release (already in master) you will have the ability to
write your own python code to render the tabs however you like.

**Reply by lokesh-krishna** [reply]
Doesn't that only let one pick between angled, slanted or round? That's what I could find in the man page and on the website. Am I missing something? I believe the separator character only applies when using the separator tab style and even then only applies to the right and not to the left of a tab.

**Reply by kovidgoyal** [reply]
You put your rounded characters into the tab title template to use them as separators.

**Reply by lokesh-krishna** [reply]
Thanks for pointing me towards that. Didn't think of using them for that. Is it possible to specify both background and foreground for the same character though? The man page example doesn't cover it and my attempts, by nesting the bg tags between the fg ones or nesting the fg ones between the bg ones aren't working. 

EDIT: My bad, figured it out. Got what I was going for :D 

![image](https://user-images.githubusercontent.com/20235646/131320580-1c128564-26d4-4cf5-b3fd-6ddee6e651d9.png)

**Reply by anuvyklack** [reply]
@lokesh-krishna How do you achieve this tabs appearance? Could you please share the config?

**Reply by lokesh-krishna** [reply]
```
tab_bar_edge    top
tab_bar_margin_width    10
tab_bar_margin_height    15 0
tab_bar_style    fade
tab_fade    0 
tab_title_template          "{fmt.bg._080808}{fmt.fg._303030}{fmt.fg.default}{fmt.bg._303030}{fmt.fg._c6c6c6} {title} {fmt.fg.default}{fmt.bg.default}{fmt.fg._303030}{fmt.fg.default}"
active_tab_title_template   "{fmt.bg._080808}{fmt.fg._80a0ff}{fmt.fg.default}{fmt.bg._80a0ff}{fmt.fg._080808} {title} {fmt.fg.default}{fmt.bg.default}{fmt.fg._80a0ff}{fmt.fg.default}"
```

**Reply by anuvyklack** [reply]
Thank you!

**Comment by kovidgoyal** [comment]
As far as I know nesting works fine, however if you are gonna get that
complex, I suggest you just wait for the next release and implement tab
drawing directly in python
